### ISSUE TITLE
Fail on truly unexpected packets, but recognize types of packets that rpgp can safely ignore

### ISSUE THREAD

**Issue by hko-s** [rpgp/rpgp#269]
Currently, the packet parsing machinery opts to accept all kinds of unexpected packets by just ignoring them. This is good for not crashing, but comes with a risk of ignoring packets that would be important to handle, and are dangerous to ignore.

Ideally, I think, unknown packets should cause parsing to stop and fail. Except in explicitly good-listed cases.
I'd propose to use `Error::Unsupported` as a mechanism for the packet parsing machinery to signal packets that it saw,  doesn't handle but understood what they are, and knows they are safe to ignore.
This could, for example, be the case for a (sub)key packet that is older or newer than what rpgp supports. Or for a signature packet that uses an algorithm that rpgp doesn't support.

All other, truly unexpected, types of packets should lead to an error, and abort the parsing process.

I've done a rough sketch of such a change, and I think it's promising.
However, this is a pretty invasive change, and I wanted to get feedback on the general direction first.

**Comment by dignifiedquire** [comment]
I agree the handling should be improved here, this was mainly done in early stages to make sure we don't reject sth just because we don't have support for it yet. Now that general support is much better we should be more strict about this.

**Comment by hko-s** [comment]
Great, then I'll clean up my experiments and prepare a concrete suggestion that we can discuss and test.

**Comment by hko-s** [comment]
This is done and has been merged via #271.
